spec/requests/incomes_spec.rb

```diff
@@ -0,0 +1,95 @@
+require 'rails_helper'
+
+RSpec.describe IncomesController, type: :request do
+  describe 'POST /assessments/:assessment_id/incomes' do
+    let(:assessment) { create :assessment }
+    let(:benefit_receipts) { attributes_for_list(:benefit_receipt, 2) }
+    let(:wage_slips) { attributes_for_list(:wage_slip, 2) }
+    let(:headers) { { 'CONTENT_TYPE' => 'application/json' } }
+    let(:date_in_future) { 3.days.from_now.strftime('%Y-%m-%d') }
+
+    let(:params) do
+      {
+        benefits: benefit_receipts,
+        wage_slips: wage_slips
+      }
+    end
+
+    subject { post assessment_income_path(assessment), params: params.to_json, headers: headers }
+
+    it 'returns http success', :show_in_doc do
+      subject
+      expect(response).to have_http_status(:success)
+    end
+
+    it 'creates wage_slips' do
+      expect { subject }.to change { assessment.wage_slips.count }.by(2)
+    end
+
+    it 'creates benefit_receipts' do
+      expect { subject }.to change { assessment.benefit_receipts.count }.by(2)
+    end
+
+    it 'sets success flag to false' do
+      subject
+      expect(parsed_response[:success]).to be true
+    end
+
+    it 'returns blank errors' do
+      subject
+      expect(parsed_response[:errors]).to be_empty
+    end
+
```

Comment:
I think we should have a spec which checks that the response contains the created benefit receipts and wage slips.